src/segment-loader.js

```diff
@@ -1461,6 +1461,42 @@ export default class SegmentLoader extends videojs.EventTarget {
         startTime: this.syncPoint_.time
       });
 
+      if (!mediaInfo) {
+        return null;
+      }
+
+      if (this.fetchAtBuffer_) {
+        const segment = segments[mediaInfo.segmentIndex];
+        const segmentEnd = mediaInfo.startTime + segment.duration;
+
+        if (segmentEnd > bufferedEnd) {
+          const difference = segmentEnd - bufferedEnd;
```

Comment:
Could there be a case where the bufferedEnd is larger than the segmentEnd? Should this be the absolute value here?

Reply:
We are fine in this case. It means that we selected the correct segment.